### ISSUE TITLE
Add an option to scale the room horizontally

### ISSUE THREAD

**Issue by evg-zhabotinsky** [alvr-org/ALVR#1052]
This is a "feature request", but I intend to implement it myself, just need a little help to get started.

The bigger the room, the better for roomscale gameplay. You can physically expand playing area only so much, but what about virtually?

**The basic idea is this:**
Scale horizontal HMD coordinates by a configurable factor (e.g. x2), while leaving its vertical coordinate and controller coordinates relative to HMD untouched.

I suspect that ALVR server would be the easiest place to insert that hack, but the code base is still huge, I don't know where to start looking.
* Where would be the right place to intercept and modify the coordinates of HMD and controllers?
* Would I also need to scale the "room dimensions" or something?

Any help is appreciated. I have no idea how my brain would handle such a movement discrepancy, but am really curious to find out.

**Comment by zarik5** [comment]
This is something that i wanted to do for a while. Currently the positional and rotation offset of controllers and headset is done on the C++ side and it's a bit messy. I would like to create a tracking_mapping.rs module in `alvr/server/src` where this kind of transformations can be applied before sending the tracking data to the OpenVR driver. Also here it should be added a way to reset the view direction (gravity locked or tilted-view) when the PlayspaceSync packet is received. In the future this module can also be used to map and calibrate trackers.
I'm currently working on something else for work but you can contact me on Discord if you need any help with this.

**Comment by evg-zhabotinsky** [comment]
Didn't have time to really look into code yet, but did some more thinking and research on the go.

It seems that there already is a "universal" hack for this kind of thing: [matzman666/OpenVR-InputEmulator](https://github.com/matzman666/OpenVR-InputEmulator) It even has been maintained somewhat recently, especially as [Dirt-e/OpenVR-MotionCompensation](https://github.com/Dirt-e/OpenVR-MotionCompensation). Seems to be a good base for coding any pose adjustments for SteamVR.

Provided that the above can be made to work alongside ALVR, does it make sense to implement the adjustments in ALVR itself?

One other reason I realized why ALVR might not be the right place is that the best reference point for this coordinate adjustment isn't HMD like I suggested above, but the feet. Referencing HMD would make you move virtually when you just lean forward while standing in place. ALVR isn't supposed to deal with FBT, at least not yet I guess? And [ju1ce/April-Tag-VR-FullBody-Tracker](https://github.com/ju1ce/April-Tag-VR-FullBody-Tracker) lets you have FBT with just your phone and some cardboard, so it doesn't seem unreasonable to prefer tracked feet as the reference.

P.S.
If the above-mentioned AptilTag tracker was packed into an Android app that streamed tag coordinates, including some tags on the Quest itself, over network, ALVR probably could use that data to provide FBT too, which would require less setup for the end user. And then the feet-referenced scaling could be done in ALVR too, ditching another SteamVR plugin. Not sure if that is a good idea, though, would either lock all that to Quest ecosystem or require maintaining the code separately from the "universal" counterparts.
